### PR TITLE
add usr/pwd params to client for authentication with etcd

### PR DESCRIPTION
**Pull request by ulaskeles**
For https://github.com/kragniz/python-etcd3/issues/98

### CODE REVIEW

**Comment by jd** [comment]
LGTM but need rebasing.

**Comment by ulaskeles** [comment]
Rebased.

File: tests/test_etcd3.py [hunk 1]
```diff
@@ -642,6 +678,35 @@ def test_compact(self, etcd):
...
+            cert_cert=None
+        )
+        assert client.uses_secure_channel is False
+
+    def test_user_pwd_auth(self):
+        auth_mock = mock.MagicMock()
+        auth_resp_mock = mock.MagicMock()
+        auth_resp_mock.token = "the_token"
+        auth_mock.Authenticate = auth_resp_mock
+        etcdrpc.AuthStub = auth_mock
```
Comment (jd) [inline tests/test_etcd3.py]: I don't see this cleaned anywhere. Wouldn't be better to use `with mock.patch …` or something like that?

**Comment by davissp14** [comment]
Is this being worked on?  This feature is in pretty high demand.

**Comment by jackic23** [comment]
Tried this PR for some work that I was doing that required username and password.  Seemed to work fine. Used to get gRPC errors about credentials and now I'm able to connect, put and get keys, as well as setup leases.

**Comment by bruceadams** [comment]
I concur with @jackic23: The code in this pull request works for connecting to `etcd` hosted on Compose.io, which requires a user & password. I just did:
```
pip install https://github.com/ulaskeles/python-etcd3/archive/auth-usr-pwd.zip
```
instead of hacking my installed packages.

I would love to see this merged into master and released.

I do not fully understand @jd's code review comment. Looking at the current code, I do not see any change that appears to address @jd's concern.

**Comment by kragniz** [comment]
Thanks all for the feedback, I'm going to merge this in now

**Pull request merged**